File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations as pm

def is_prime(n):
    if n <=1:
        return False
    else:
        for i in range(2,int(n**0.5+1)):
            if n % i == 0:
                return False
        return True


def solution(numbers):
    chk = {}
    answer = 0
    cnt = 1
    while cnt < len(numbers)+1:
        for i in pm(numbers,cnt):
            tmp = "".join(i)
            if tmp[0] == 0:
                pass
            if is_prime(int(tmp)):
                if int(tmp) in chk.keys():
                    pass
                else:
                    chk[int(tmp)] = 1
                    answer +=1
        cnt+=1
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
// 주어진 숫자들로 만들 수 있는 모든 숫자 구하기

function solution(numbers) {
    const arr = [];
    const rst = [];
    
    const isPrime = (n) => {
        for(let i =2; i < n; i++){
            if(n % i === 0){
                return false;
            }
        }
        return n > 1;
    }
    
    const go = (cur,cnt,visited) => {
        if(cur.length > cnt){
            const tmp = Number(cur.join(""));
            if(isPrime(tmp)){
                if(!rst.includes(tmp)){
                	rst.push(tmp);
                }
            }
            return;
        }
        
        for(let i =0; i < arr.length; i++){
            if(!visited[i]){
                cur.push(arr[i]);
                visited[i] = true;
                go(cur, cnt, visited);
                cur.pop();
                visited[i] = false;
            }
        }
    }
    
    // string to array
    for(const chr of numbers){
        arr.push(chr);
    }
    // console.log(arr);
    
    for(let i=0; i<arr.length; i++){
        const currentLength = i;
        const visited = Array(arr.length).fill(false);
        go([],i, visited);
        
    }
    
    return rst.length;
}